### ISSUE TITLE
What does it mean when `hsl` is `None` in the callbacks?

### ISSUE THREAD

**Issue by rec** [rec/colorsphere#4]
Example [here](https://github.com/rec/colorsphere/blob/92edc0df67524b44cda19c385c09072bce3e0703/colorsphere/xled_launcher.py#L36-L48).

Is this a callback from outside the sphere?

Perhaps this is actually a different _kind_ of callback, then?

**Comment by Anders-Holst** [comment]
Yes, when you leave the sphere, there is no color under the pointer, and hsl is set to None accordingly. The same holds for the click callback. Originally, the callback was not called at all when the pointer was outside of the sphere. However, I wanted it to do things when leaving the sphere (specifically, leaving rt mode by returning to the previous mode or the last set color), and the easiest way was to remove the check on hsl previously done before the callback, and add it inside the callback instead.

**Comment by Anders-Holst** [comment]
Aha, what you suggest is to add a new callback which is called when leaving/entering the sphere. Yea one can do that. Build an entire event system with lots of event handlers if one likes it... I tried to keep it simple and straightforward, but agree that it is at the cost of overloading this callback to mean different things.
